src/components/forms/UserReference/index.ts

```diff
@@ -1,186 +1,273 @@
         // referenced users won't be available, so get user details from dx api
         const { getOperatorDetails } = PCore.getUserApi();
-        getOperatorDetails(userId).then((resp) => {
+        getOperatorDetails(this.userID).then((resp) => {
           if (
             resp.data &&
             resp.data.pyOperatorInfo &&
             resp.data.pyOperatorInfo.pyUserName
           ) {
-            //setUserName(res.data.pyOperatorInfo.pyUserName);
             this.userName = resp.data.pyOperatorInfo.pyUserName;
           }
         });
       }
-    } else if (displayAs === DROPDOWN_LIST) {
-      const queryPayload = {
-        dataViewName: OPERATORS_DP
-      };
+    }
+  }
 
-      // PCore.getRestClient()
-      //   .invokeRestApi("getListData", { queryPayload })
-      //   .then((resp) => {
-      //     const ddDataSource = resp.data.data.map((listItem) => ({
-      //       key: listItem.pyUserIdentifier,
-      //       value: listItem.pyUserName
-      //     }));
-      //     setDropDownDataSource(ddDataSource);
-      //     setLoading(false);
-      //   })
-      //   .catch((err) => {
-      //     console.log(err);
-      //   });
+  theRenderedDiv() {
+    if (this.bDebug) {
+      console.log(`${this.theComponentName}: theRenderedDiv`);
     }
+    return html`${this.bVisible
+      ? html`
+          <div class="psdk-user-reference">
+            <case-operator-widget
+              class="psdk-double"
+              .pConn=${this.thePConn}
+              name=${this.userName}
+              label=${this.label}
+              theId=${this.userID}
+            ></case-operator-widget>
+          </div>
+        `
+      : nothing}`;
+  }
 
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (name === "dropDownDataSource") {
+      if (oldValue !== newValue) {
+        this.getData();
+      }
+    }
   }
 
+  getData() {
+    if (
+      this.displayAs === this.DROPDOWN_LIST ||
+      this.displayAs === this.SEARCH_BOX
+    ) {
+      const queryPayload = {
+        dataViewName: this.OPERATORS_DP,
+      };
 
-  theRenderedDiv() {
-    if (this.bDebug) { console.log(`${this.theComponentName}: theRenderedDiv`); }
-    return html`${ this.bVisible ?
-      html`
-      <div class="form-group psdk-user-reference">
-        <case-operator-widget class="psdk-double" .pConn=${this.thePConn} name=${this.userName} label=${this.label} theId=${this.userID}></case-operator-widget>
-        </div>
-      `
-      :
-      nothing
-      }`;
+      PCore.getRestClient()
+        .invokeRestApi("getListData", { queryPayload })
+        .then((resp) => {
+          const ddDataSource = resp.data.data.map((listItem) => ({
+            key: listItem.pyUserIdentifier,
+            value: listItem.pyUserName,
+          }));
+          this.dropDownDataSource = ddDataSource;
+          this.dropDownDataSource = JSON.stringify(this.dropDownDataSource);
```

Comment:
why are you changing it as string?

Reply:
I was passing the array before but that didn't work then I'd to convert it to a string first in parent then parse it again in the child component.